### PR TITLE
Record volume material in the Interactor

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
For the Volume material mapping, material interaction in volume need to be added to the result of Material Interactor. this is onlu performed if recordInteractions is true.

### CODE REVIEW

File: Core/include/Acts/Propagator/MaterialInteractor.hpp [hunk 1]
```diff
@@ -159,12 +188,56 @@ struct MaterialInteractor {
...
+    mi.time = d.time;
+    mi.direction = d.dir;
+    mi.sigmaPhi2 = d.variancePhi;
+    mi.sigmaTheta2 = d.varianceTheta;
+    mi.sigmaQoP2 = d.varianceQoverP;
+    mi.surface = nullptr;
+    mi.volume = d.volume;
+    mi.pathCorrection = d.pathCorrection;
+    mi.materialProperties = d.slab;
+    result.materialInteractions.push_back(std::move(mi));
```
[the remark this answers is not in the document]
Comment (Corentin-Allaire) [reply Core/include/Acts/Propagator/MaterialInteractor.hpp:216]: The way I wrote this it only does a push back if recordInteractions == true and from what I understand this variable is false by default and only switch to true when doing the material recording. So what I added only result to one (or two) extra if check in the non material recording case.

File: Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  double pathCorrection;
+
+  double variancePhi = 0.;
+  double varianceTheta = 0.;
+  double varianceQoverP = 0.;
+
+  double Eloss = 0.;
+  double nextP;
+
+  /// @brief Contructor
```
[the remark this answers is not in the document]
Comment (asalzburger) [reply Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp:46]: @Corentin-Allaire  can you fix the typo?

[the remark this answers is not in the document]
Comment (Corentin-Allaire) [reply Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp]: Fixed

**Comment by asalzburger** [comment]
@Corentin-Allaire there's a tiny change in coverage, but this was not conclusive for me where this is from. I am merging this.

**Pull request merged**